Userland/Games/BrickGame/BrickGame.cpp

```diff
@@ -0,0 +1,511 @@
+        LevelMap { 600000, AK::Time::from_nanoseconds(3000 / 60) },
+        LevelMap { 700000, AK::Time::from_nanoseconds(2000 / 60) },
+        LevelMap { 10000000, AK::Time::from_nanoseconds(1000 / 60) }
+    };
+
+    void set_current_block(Block const& block)
+    {
+        if (!block.has_collision(m_well)) {
+            m_block = block;
+        }
+    }
+
+public:
+    Bricks() { reset(); }
+
+    [[nodiscard]] static constexpr size_t width() { return Well::num_columns(); }
+    [[nodiscard]] static constexpr size_t height() { return Well::num_rows(); }
+
+    [[nodiscard]] bool is_game_over() const { return m_is_game_over; }
+    [[nodiscard]] unsigned score() const { return m_score; }
+    [[nodiscard]] unsigned level() const { return m_level; }
+
+    [[nodiscard]] bool new_block()
+    {
+        m_block = Block().random_shape();
+        m_is_game_over = m_block.has_collision(m_well);
+        return !m_is_game_over;
+    }
+    [[nodiscard]] bool operator[](Position pos) const
+    {
+        return m_well[pos] || m_block[pos];
+    }
+
+    void rotate_left() { set_current_block(Block(m_block).rotate_left()); }
+    void rotate_right() { set_current_block(Block(m_block).rotate_right()); }
+
+    void move_left() { set_current_block(Block(m_block).move_left()); }
+    void move_right() { set_current_block(Block(m_block).move_right()); }
+    bool move_down()
+    {
+        auto const block = Block(m_block).move_down();
+        if (block.has_collision(m_well)) {
+            m_block.place_into(m_well);
+            return new_block();
+        }
+        m_block = block;
+        return true;
+    }
+    bool move_down_fast()
+    {
+        for (auto block = m_block;; block.move_down()) {
+            if (block.has_collision(m_well)) {
+                m_block.place_into(m_well);
+                return new_block();
+            }
+            m_block = block;
+        }
+    }
+    void remove_line(size_t line) { m_well.remove_line(line); }
+
+    void check_lines()
+    {
+        auto num_removed_rows { 0 };
+        for (int row = int(Well::num_rows() - Well::bottom_margin()); row >= 0;
+             --row) {
+            if (m_well.is_complete(row)) {
+                remove_line(row);
+                ++num_removed_rows;
+                ++row;
+            }
+        }
+        switch (num_removed_rows) {
+        case 1:
+            m_score += 40 * (m_level + 1);
+            break;
+        case 2:
+            m_score += 100 * (m_level + 1);
+            break;
+        case 3:
+            m_score += 300 * (m_level + 1);
+            break;
+        case 4:
+            m_score += 1200 * (m_level + 1);
+            break;
+        default:
+            break;
+        }
+    }
+
+    void update()
+    {
+        for (size_t i = 0; i < s_level_map.size(); i++) {
+            if (m_score >= s_level_map[i].m_score) {
+                m_level = i;
+            } else
+                break;
+        }
```

Comment:
Weird curly braces here -- regardless, as this `if` statement should be without them